basemq/tasks/FairMQProcessorTask.h

```diff
@@ -21,16 +21,16 @@
 class FairMQProcessorTask : public FairTask
 {
   public:
-    FairMQProcessorTask();
+    FairMQProcessorTask() = default;
     FairMQProcessorTask(const FairMQProcessorTask&) = delete;
     FairMQProcessorTask operator=(const FairMQProcessorTask&) = delete;
 
-    virtual ~FairMQProcessorTask();
+    virtual ~FairMQProcessorTask() = default;
 
-    virtual void Exec(Option_t* opt = "0");
+    virtual void Exec(Option_t* opt = "0") {}
```

Comment:
There is actually a small difference - here the `opt` parameter is defaulted, while in `TTask` it is not. So removing it here would require modifying the call in `FairMQProcessor`. Can't think of a good reason for one (leaving this line as is) or the other (modifying FairMQProcessor). You?

Of course there is also a chance that this class is used, so removing it would be breaking. Found one instance in https://git.panda.gsi.de/PandaRootGroup/PandaRoot/-/blob/dev/detectors/mvd/MvdOfflineTBAnalysis_Topix4/MQ/PndMQTopix4ProcessorTask.h, but there it is overriden again... :)

Reply:
The function was added 2015 with c3cc5074a0756466b9b766c5e2f09ac0fcab375b according to the commit message to remove compiler warnings.

I would propose to change it to

`virtual void Exec(Option_t* ) {}`

which does nothing and also has no problems with parameters. In my opinion the default parameter was added to avoid a warning about unused parameters.

Reply:
@fuhlig1 Changing to that would be the same as removing that line, which I would then rather do, if we are ok with that little breaking change.

Reply:
In my opinion it is not breaking since it was never used.

Reply:
Fine with me. I'll remove the line then.